node/channel.js

```diff
@@ -453,20 +454,37 @@ TChannel.prototype.makeSubChannel = function makeSubChannel(options) {
     return chan;
 };
 
-TChannel.prototype.listen = function listen(port, host, callback) {
+TChannel.prototype.listen = function listen(port, host, options, callback) {
     // Note:
     // - 0 is a valid port number, indicating that the system must assign an
     //   available ephemeral port
     // - 127.0.0.1 is a valid host, primarily for testing
+    // - options supports:
+    //   {fd: X} - listen will attach to the existing socket, and host/port
+    //   remain required, but become only descriptive. See also:
+    //   https://nodejs.org/api/net.html#net_server_listen_handle_callback
     var self = this;
+    if (options && typeof options === 'function') {
+        callback = options;
+        options = {};
+    }
+    options = options || {};
     assert(!self.listened, 'TChannel can only listen once');
     assert(typeof host === 'string', 'TChannel requires host argument');
     assert(typeof port === 'number', 'TChannel must listen with numeric port');
+    assert(options.fd === undefined || typeof options.fd === 'number', 'TChannel listen options.fd must be numeric');
```

Comment:
good question. i think the level of validation is partially subjective. It looks like the `port` check above doesn't account for NaN. and you may ask, "why should i?". that'd be a good question to ask too ;). in my mind:

1) make sure listen() doesn't blow up with the input that is provided to it.
2) make sure there's reasonable behavior for the developer developing against the TChannel API
3) don't go farther than you need to go to protect said developer from shooting themselves in the foot.